protobuf/test/list_equality_test.dart

```diff
@@ -0,0 +1,52 @@
+// Test for ensuring that protobuf lists compare using value semantics.
+library list_equality_test;
+
+import 'package:test/test.dart';
+
+import 'mock_util.dart' show T;
+
+void main() {
+  test('empty lists compare as equal', () {
+    final first = T();
+    final second = T();
+    expect(first.int32s == second.int32s, isTrue);
+  });
+
+  test('empty frozen lists compare as equal', () {
```

Comment:
The rest of the test cases are for completeness. This is the test that was failing before this change